Compilation error in Arch and derivatives (AUR package) due to cmake 4

Arch and derivatives have cmake 4, which has deprecated compatibility with cmake < 3.5

```
CMake Error at 3rdparty/json/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
```

With the patch this progresses further, but I'm still getting cmake related errors:

`-- Protobuf_ROOT=/home/leo/build/openjkdf2-git/src/openjkdf2-git-build/protobuf
CMake Warning at /usr/share/cmake/Modules/ExternalProject.cmake:1498 (message):
  Line 'CMAKE_POSITION_INDEPENDENT_CODE=ON' does not match regex.  Ignoring.
Call Stack (most recent call first):
  /usr/share/cmake/Modules/ExternalProject.cmake:2591 (_ep_command_line_to_initial_cache)
  /usr/share/cmake/Modules/ExternalProject.cmake:2638 (_ep_extract_configure_command)
  /usr/share/cmake/Modules/ExternalProject.cmake:3079 (_ep_add_configure_command)
  cmake_modules/build_gns.cmake:29 (ExternalProject_Add)
  cmake_modules/config_platform_deps.cmake:104 (include)
  CMakeLists.txt:206 (include)


-- Going to build “libpng 1.6.39” from Git module
-- Going to build “SDL 2.26.5” from Git module
SDL_VERSION_MAJOR == 2
SDL_VERSION_MINOR == 26
SDL_VERSION_PATCH == 5
-- Going to build “SDL_mixer 2.6.3” from Git module
SDL_MIXER_VERSION_MAJOR == 2
SDL_MIXER_VERSION_MINOR == 6
SDL_MIXER_VERSION_PATCH == 3
-- Found PkgConfig: /usr/bin/pkg-config (found version "2.4.3")
-- Checking for module 'gtk+-3.0'
--   Found gtk+-3.0, version 3.24.49
-- Checking for module 'libcurl'
--   Found libcurl, version 8.13.0
-- Configuring done (0.7s)
-- Generating done (0.0s)
-- Build files have been written to: /home/leo/build/openjkdf2-git/src/openjkdf2-git-build
==> Starting pkgver()...
==> Starting build()...
[  1%] Creating directories for 'GLEW'
[  1%] No download step for 'GLEW'
[  2%] No update_disconnected step for 'GLEW'
[  2%] No patch_disconnected step for 'GLEW'
[  3%] Performing configure step for 'GLEW'
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
make[2]: *** [CMakeFiles/GLEW.dir/build.make:92: GLEW-prefix/src/GLEW-stamp/GLEW-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:137: CMakeFiles/GLEW.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...`

I was able to make more progress by adding `export CMAKE_POLICY_VERSION_MINIMUM=3.5` before calling cmake`, to apply to all the library builds.

But now I'm running into the error:

```
...
-- Configuring vorbis 1.3.7
-- Configuring done (8.1s)
CMake Error in external/vorbis/lib/CMakeLists.txt:
  export called with target "vorbis" which requires target "ogg" that is not
  in any export set.


-- Generating done (0.2s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
make[3]: *** [CMakeFiles/SDL_mixer.dir/build.make:93: SDL_mixer-prefix/src/SDL_mixer-stamp/SDL_mixer-configure] Error 1
make[2]: *** [CMakeFiles/Makefile2:361: CMakeFiles/SDL_mixer.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[100%] Linking C static library lib/libglew.a
create libGLEW symbolic link
[100%] Built target libglew_static
[ 22%] Performing install step for 'GLEW'
[100%] Built target libglew_static
Install the project...
-- Install configuration: "Release"
-- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/lib/cmake/glew/glewConfig.cmake
-- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/lib/cmake/glew/glewConfig-release.cmake
-- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/include/GL/eglew.h
-- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/include/GL/glew.h
-- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/include/GL/glxew.h
-- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/include/GL/wglew.h
[ 22%] Completed 'GLEW'
[ 22%] Built target GLEW
make[1]: *** [CMakeFiles/Makefile2:479: CMakeFiles/openjkdf2.dir/rule] Error 2
make: *** [Makefile:254: openjkdf2] Error 2
```

I am stuck at the same point, did you manage to get any further?

> I was able to make more progress by adding `export CMAKE_POLICY_VERSION_MINIMUM=3.5` before calling cmake`, to apply to all the library builds.
> 
> But now I'm running into the error:
> 
> ```
> ...
> -- Configuring vorbis 1.3.7
> -- Configuring done (8.1s)
> CMake Error in external/vorbis/lib/CMakeLists.txt:
>   export called with target "vorbis" which requires target "ogg" that is not
>   in any export set.
> 
> 
> -- Generating done (0.2s)
> CMake Generate step failed.  Build files cannot be regenerated correctly.
> make[3]: *** [CMakeFiles/SDL_mixer.dir/build.make:93: SDL_mixer-prefix/src/SDL_mixer-stamp/SDL_mixer-configure] Error 1
> make[2]: *** [CMakeFiles/Makefile2:361: CMakeFiles/SDL_mixer.dir/all] Error 2
> make[2]: *** Waiting for unfinished jobs....
> [100%] Linking C static library lib/libglew.a
> create libGLEW symbolic link
> [100%] Built target libglew_static
> [ 22%] Performing install step for 'GLEW'
> [100%] Built target libglew_static
> Install the project...
> -- Install configuration: "Release"
> -- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/lib/cmake/glew/glewConfig.cmake
> -- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/lib/cmake/glew/glewConfig-release.cmake
> -- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/include/GL/eglew.h
> -- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/include/GL/glew.h
> -- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/include/GL/glxew.h
> -- Installing: /home/jkohler/Projects/JKMots/OpenJKDF2/build_linux64/glew/include/GL/wglew.h
> [ 22%] Completed 'GLEW'
> [ 22%] Built target GLEW
> make[1]: *** [CMakeFiles/Makefile2:479: CMakeFiles/openjkdf2.dir/rule] Error 2
> make: *** [Makefile:254: openjkdf2] Error 2
> ```

Currently, this is where it fails for me:

```
-- Going to build “SDL 2.26.5” from Git module
SDL_VERSION_MAJOR == 2
SDL_VERSION_MINOR == 26
SDL_VERSION_PATCH == 5
-- Going to build “SDL_mixer 2.6.3” from Git module
SDL_MIXER_VERSION_MAJOR == 2
SDL_MIXER_VERSION_MINOR == 6
SDL_MIXER_VERSION_PATCH == 3
-- Using system OpenAL
-- Found PkgConfig: /usr/bin/pkg-config (found version "2.5.1")
-- Checking for module 'gtk+-3.0'
--   Found gtk+-3.0, version 3.24.51
-- Checking for module 'libcurl'
--   Found libcurl, version 8.17.0
-- Configuring done (0.8s)
CMake Error at cmake_modules/config_platform_deps.cmake:327 (add_dependencies):
  The dependency target "PROTOBUF" of target "GNS_HACK_ZLIB" does not exist.
Call Stack (most recent call first):
  CMakeLists.txt:219 (include)


-- Generating done (0.0s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
==> ERROR: A failure occurred in prepare().
    Aborting...
error: failed to build 'openjkdf2-git-0.9.6.r87.g1baa1e8-1': 
error: packages failed to build: openjkdf2-git-0.9.6.r87.g1baa1e8-1
```

This is the reason why I opened https://github.com/shinyquagsire23/OpenJKDF2/issues/389
I already have all dependencies built in my OS, so it seems silly to rebuild them all for this.

--- a/cmake_modules/build_gns.cmake
+++ b/cmake_modules/build_gns.cmake
@@ -46,6 +46,7 @@ ExternalProject_Add(
                            -DProtobuf_USE_STATIC_LIBS:BOOL=TRUE
                            -DProtobuf_ROOT:PATH=${Protobuf_ROOT}
                            ${GAMENETWORKINGSOCKETS_PROTOC_EXECUTABLE}
+                           -DCMAKE_POLICY_VERSION_MINIMUM=3.5
                            ${GAMENETWORKINGSOCKETS_EXTRA_ARGS}
     DEPENDS                PROTOBUF ${GAMENETWORKINGSOCKETS_DEPENDS}
     PATCH_COMMAND          git restore CMakeLists.txt src/CMakeLists.txt &&

--- a/cmake_modules/build_host_zlib.cmake
+++ b/cmake_modules/build_host_zlib.cmake
@@ -36,6 +36,7 @@ ExternalProject_Add(
                         ${ZLIB_HOST_TOOLCHAIN_ARG}
                         -DCMAKE_INSTALL_LIBDIR=lib
                         -DCMAKE_BUILD_TYPE:STRING=Release
+                        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
     BUILD_BYPRODUCTS    ${ZLIB_HOST_STATIC_LIBRARY_PATH} ${ZLIB_HOST_DYNAMIC_LIBRARY_PATH} ${ZLIB_HOST_IMPORT_LIBRARY_PATH}
 )
 

--- a/cmake_modules/build_libpng.cmake
+++ b/cmake_modules/build_libpng.cmake
@@ -33,6 +33,7 @@ ExternalProject_Add(
                         -DPNG_STATIC:BOOL=TRUE
                         -DPNG_EXECUTABLES:BOOL=FALSE
                         -DPNG_TESTS:BOOL=FALSE
+                        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
                         -DZLIB_ROOT:PATH=${ZLIB_ROOT}
     DEPENDS             ${LIBPNG_DEPENDS}
     BUILD_BYPRODUCTS    ${PNG_STATIC_LIBRARY_PATH} #${PNG_IMPORT_LIBRARY_PATH}

--- a/cmake_modules/build_protobuf.cmake
+++ b/cmake_modules/build_protobuf.cmake
@@ -126,6 +126,7 @@ ExternalProject_Add(
                            -Dprotobuf_BUILD_EXAMPLES:BOOL=FALSE
                            -Dprotobuf_BUILD_PROTOC_BINARIES:BOOL=${PROTOBUF_BUILD_PROTOC_BINARIES}
                            -Dprotobuf_DISABLE_RTTI:BOOL=TRUE
+                           -DCMAKE_POLICY_VERSION_MINIMUM=3.5
                            -DZLIB_ROOT:PATH=${ZLIB_ROOT}
     DEPENDS                ${PROTOBUF_DEPENDS} # add platform specific zlib depency
     BUILD_BYPRODUCTS       ${libprotobuf_STATIC_LIBRARY_PATH} ${libprotobuf_SHARED_LIBRARY_PATH}

--- a/cmake_modules/build_protoc.cmake
+++ b/cmake_modules/build_protoc.cmake
@@ -16,6 +16,7 @@ ExternalProject_Add(
                         -Dprotobuf_BUILD_EXAMPLES:BOOL=FALSE
                         -Dprotobuf_BUILD_PROTOC_BINARIES:BOOL=TRUE
                         -Dprotobuf_DISABLE_RTTI:BOOL=TRUE
+                        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
                         -DZLIB_ROOT:PATH=${ZLIB_HOST_ROOT}
     BUILD_COMMAND       ${CMAKE_MAKE_PROGRAM} protoc
     DEPENDS             ${PROTOC_DEPENDS}

